src/internal_events/aws_s3.rs

```diff
@@ -44,18 +44,18 @@ pub mod source {
     impl<'a> InternalEvent for SqsMessageReceiveError<'a> {
         fn emit_logs(&self) {
             error!(
-                message = "Failed to fetch SQS events.",
-                error = %self.error,
-                error_type = "request_failed",
+                message = %format!("Failed to fetch SQS events: {:?}", self.error),
```

Comment:
I'm not excited about this pattern. Part of the point of having structured fields in the error output is to avoid having to do this kind of formatting.

Reply:
I think this was motivated by https://github.com/vectordotdev/vector/pull/11317 since I think `error` should really be an "error code" or some other limited set of values since it is also used as a metric tag.

We could split it off as a separate field like `error_description` though. Or keep `error` and add `error_code` where `error` only appears in log messages and `error_code` appears in both.